app/vmselect/prometheus/prometheus.go

```diff
@@ -76,6 +76,7 @@ func FederateHandler(startTime time.Time, w http.ResponseWriter, r *http.Request
 	if err != nil {
 		return err
 	}
+	defer federateDuration.UpdateDuration(startTime)
```

Comment:
I think it is better from readability and maintainability PoV moving these `defer` calls to the first line of every measured function.

Reply:
This will account for queries which fail the validation. Do you think it matters?

Reply:
I think this is OK

Reply:
updated